src/index.ts

```diff
@@ -114,6 +105,30 @@ function resolveSchema(maybeSchema: ZodAny | { properties: ZodAny }): Pick<ZodAn
   throw new Error(`Invalid schema passed: ${JSON.stringify(maybeSchema)}`);
 }
 
+const ajv = new Ajv({
+  removeAdditional: 'all',
+  useDefaults: true,
+  coerceTypes: 'array',
+});
+
+//@ts-ignore
+export const validatorCompiler: FastifySchemaCompiler<ZodAny> = ({ schema }) => {
+  //@ts-ignore
+  if (schema.safeParse) {
+    return (data) => {
+      try {
+        schema.parse(data);
+        return { value: data };
+      } catch (e) {
+        const validationError = fromZodError(e as ZodError);
+        return { error: validationError };
+      }
+    };
+  }
+
+  return ajv.compile(schema);
```

Comment:
I don't think this is correct. You are basically assuming that sometimes you are going to get an ajv schema, and hardcode falling back to using it if there are no zod-specific methods found. This is brittle, and too much hardcoding.

I don't mind a more universal approach that would allow users to pass their own fallback compilers for when zod one is not found, but it shouldn't be hardcoded

Reply:
Agree with you, but I will keep the default if no fallback is precised. Because by default ajv is used on fastify.